From ec4efe710859b3e01c90f96ebef91e9a0bc36083 Mon Sep 17 00:00:00 2001
From: Stefan Agner <stefan.agner@toradex.com>
Date: Thu, 7 Nov 2019 10:37:39 +0100
Subject: [PATCH 1/2] Revert "Revert "MLK-18433 PCI: imx: remove the lpcg_xxx
 clocks in driver""

This reverts commit c1e7d4fc8c1b9bd599eed9cf848eff3ed9a34637.

This commit is required to make PCIe on Ixora working.
---
 .../boot/dts/freescale/fsl-imx8qm-device.dtsi |  6 ++----
 drivers/pci/dwc/pci-imx6.c                    | 19 +------------------
 2 files changed, 3 insertions(+), 22 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
index a804eb15b841..ea5c6eef50e1 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-device.dtsi
@@ -4245,9 +4245,8 @@
 		clocks = <&clk IMX8QM_HSIO_PCIE_A_MSTR_AXI_CLK>,
 			 <&clk IMX8QM_HSIO_PCIE_A_SLV_AXI_CLK>,
 			 <&clk IMX8QM_HSIO_PHY_X2_PCLK_0>,
-			 <&clk IMX8QM_HSIO_PCIE_X2_PER_CLK>,
 			 <&clk IMX8QM_HSIO_PCIE_A_DBI_AXI_CLK>;
-		clock-names = "pcie", "pcie_bus", "pcie_phy", "pcie_per", "pcie_inbound_axi";
+		clock-names = "pcie", "pcie_bus", "pcie_phy", "pcie_inbound_axi";
 
 		interrupt-map-mask = <0 0 0 0x7>;
 		interrupt-map =  <0 0 0 1 &gic GIC_SPI 73 IRQ_TYPE_LEVEL_HIGH>,
@@ -4289,9 +4288,8 @@
 		clocks = <&clk IMX8QM_HSIO_PCIE_B_MSTR_AXI_CLK>,
 			 <&clk IMX8QM_HSIO_PCIE_B_SLV_AXI_CLK>,
 			 <&clk IMX8QM_HSIO_PHY_X2_PCLK_1>,
-			 <&clk IMX8QM_HSIO_PCIE_X1_PER_CLK>,
 			 <&clk IMX8QM_HSIO_PCIE_B_DBI_AXI_CLK>;
-		clock-names = "pcie", "pcie_bus", "pcie_phy", "pcie_per", "pcie_inbound_axi";
+		clock-names = "pcie", "pcie_bus", "pcie_phy", "pcie_inbound_axi";
 
 		interrupt-map-mask = <0 0 0 0x7>;
 		interrupt-map =  <0 0 0 1 &gic GIC_SPI 105 IRQ_TYPE_LEVEL_HIGH>,
diff --git a/drivers/pci/dwc/pci-imx6.c b/drivers/pci/dwc/pci-imx6.c
index 5ecae3b160be..7855ded3102f 100644
--- a/drivers/pci/dwc/pci-imx6.c
+++ b/drivers/pci/dwc/pci-imx6.c
@@ -78,7 +78,6 @@ struct imx_pcie {
 	struct clk		*pcie_bus;
 	struct clk		*pcie_phy;
 	struct clk		*pcie_inbound_axi;
-	struct clk		*pcie_per;
 	struct clk		*pcie;
 	struct clk		*pcie_ext;
 	struct clk		*pcie_ext_src;
@@ -641,17 +640,8 @@ static int imx_pcie_enable_ref_clk(struct imx_pcie *imx_pcie)
 	case IMX8QXP:
 	case IMX8QM:
 		ret = clk_prepare_enable(imx_pcie->pcie_inbound_axi);
-		if (ret) {
+		if (ret)
 			dev_err(dev, "unable to enable pcie_axi clock\n");
-			break;
-		}
-		ret = clk_prepare_enable(imx_pcie->pcie_per);
-		if (ret) {
-			dev_err(dev, "unable to enable pcie_per clock\n");
-			clk_disable_unprepare(imx_pcie->pcie_inbound_axi);
-			break;
-		}
-
 		break;
 	}
 
@@ -1509,7 +1499,6 @@ static void pci_imx_clk_disable(struct device *dev)
 		break;
 	case IMX8QXP:
 	case IMX8QM:
-		clk_disable_unprepare(imx_pcie->pcie_per);
 		clk_disable_unprepare(imx_pcie->pcie_inbound_axi);
 		break;
 	}
@@ -2507,12 +2496,6 @@ static int imx_pcie_probe(struct platform_device *pdev)
 			 ("fsl,imx6sx-iomuxc-gpr");
 	} else if (imx_pcie->variant == IMX8QM
 			|| imx_pcie->variant == IMX8QXP) {
-		imx_pcie->pcie_per = devm_clk_get(dev, "pcie_per");
-		if (IS_ERR(imx_pcie->pcie_per)) {
-			dev_err(dev, "pcie_per clock source missing or invalid\n");
-			return PTR_ERR(imx_pcie->pcie_per);
-		}
-
 		imx_pcie->iomuxc_gpr =
 			 syscon_regmap_lookup_by_phandle(node, "hsio");
 		imx_pcie->pcie_inbound_axi = devm_clk_get(&pdev->dev,
-- 
2.23.0

